diff --git a/cryptol-remote-api/cryptol-remote-api.cabal b/cryptol-remote-api/cryptol-remote-api.cabal
index 6b29e31d..79e75673 100644
--- a/cryptol-remote-api/cryptol-remote-api.cabal
+++ b/cryptol-remote-api/cryptol-remote-api.cabal
@@ -80,7 +80,7 @@ executable cryptol-remote-api
   main-is:             Main.hs
   hs-source-dirs:      cryptol-remote-api
   ghc-options:
-    -threaded -rtsopts -with-rtsopts=-xb0x200000000
+    -rtsopts -with-rtsopts=-xb0x200000000
 
   build-depends:
     cryptol-remote-api,
@@ -94,7 +94,7 @@ executable cryptol-eval-server
   main-is:             Main.hs
   hs-source-dirs:      cryptol-eval-server
   ghc-options:
-    -threaded -rtsopts
+    -rtsopts
 
   build-depends:
     cryptol-remote-api,
